### PR TITLE
Remove duplicates in person search result

### PR DESCRIPTION
**Pull request by nbittich**
OP-1265 => fix Sofie's comment

### CODE REVIEW

File: app/components/person-search-by-name.js [hunk 1]
```diff
@@ -26,20 +26,28 @@ export default class PersonSearchByNameComponent extends Component {
...
+      );
     if (searchParams.trim() !== '' && result) {
       let param_object = {
         family_name: '',
         given_name: '',
       };
       param_object[fieldName] = searchParams.trim();
-      return [...[param_object], ...result];
+      const arr = [...[param_object], ...result];
+      console.log(arr);
```
Comment (claire-lovisa) [inline app/components/person-search-by-name.js:49]: This one should probably be removed ? :p

File: app/components/person-search-by-name.js [hunk 2]
```diff
@@ -26,20 +26,28 @@ export default class PersonSearchByNameComponent extends Component {
...
-
-    result = new Set(result.toArray());
+    result = result
+      .toArray()
+      .filter(
+        (v, i, a) =>
+          a.findIndex((v2) =>
+            ['family_name', 'given_name'].every((k) => v2[k] === v[k])
+          ) === i
+      );
```
Comment (claire-lovisa) [inline app/components/person-search-by-name.js:41]: Can you maybe add a comment about what this filter does ? It's not straightforward when reading it :)

**Review by claire-lovisa — APPROVED** [review]
Little comments, but the PR works as it should ! :)

File: app/components/person-search-by-name.js (hunk 2 above)
Comment (nbittich) [reply app/components/person-search-by-name.js]: indeed! I extracted the code in a separate function and added a documentation to it

File: app/components/person-search-by-name.js (hunk 1 above)
Comment (nbittich) [reply app/components/person-search-by-name.js]: oops 😱

**Pull request merged**